Multiple CI build failures due to missing syscalls/demand_paging.h

**Describe the bug**
CI on `main` shows multiple build failures due to missing `syscalls/demand_paging.h`. Git bisect shows that this was introduced in commit c972ef1a0f82d254c9a8ff2437b73ebe368ba9c4.

See e.g. https://github.com/zephyrproject-rtos/zephyr/actions/runs/6933236823/job/18859012989

**To Reproduce**
Steps to reproduce the behavior:
1. `west twister -p unit_testing -s tests/bluetooth/host/id/bt_id_adv_random_addr_check/bluetooth.host.bt_id_adv_random_addr_check.observer_scan_with_identity_enabled`
2. See error

**Expected behavior**
CI builds successfully.

**Impact**
Multiple CI build failures on `main`.

**Logs and console output**
```
[6/40] Building C object CMakeFiles/testbinary.dir/__w/zephyr/zephyr/subsys/bluetooth/host/id.c.o
FAILED: CMakeFiles/testbinary.dir/__w/zephyr/zephyr/subsys/bluetooth/host/id.c.o 
/usr/bin/gcc -DZTEST_UNITTEST -I/__w/zephyr/zephyr/tests/bluetooth/host/id/mocks -I/__w/zephyr/zephyr/subsys/testsuite/ztest/include/zephyr -I/__w/zephyr/zephyr/subsys/testsuite/include/zephyr -I/__w/zephyr/zephyr/subsys/testsuite/ztest/include -I/__w/zephyr/zephyr/subsys/testsuite/include -I/__w/zephyr/zephyr/include/zephyr -I/__w/zephyr/zephyr/include -I/__w/zephyr/zephyr/. -I/__w/zephyr/zephyr/tests/bluetooth/host/id/mocks/.. -I/__w/zephyr/zephyr/subsys/bluetooth -I/__w/zephyr/zephyr/subsys/bluetooth/host -I/__w/zephyr/zephyr/tests/bluetooth/host -m32 -Wno-error=format-zero-length -Wno-format-zero-length -imacros /__w/zephyr/zephyr/twister-out/unit_testing/tests/bluetooth/host/id/bt_id_add/bluetooth.host.bt_id_add.broadcaster_ext_adv_enabled/zephyr/include/generated/autoconf.h -Wall -I /__w/zephyr/zephyr/twister-out/unit_testing/tests/bluetooth/host/id/bt_id_add/bluetooth.host.bt_id_add.broadcaster_ext_adv_enabled/zephyr/include/generated -include ztest.h -MD -MT CMakeFiles/testbinary.dir/__w/zephyr/zephyr/subsys/bluetooth/host/id.c.o -MF CMakeFiles/testbinary.dir/__w/zephyr/zephyr/subsys/bluetooth/host/id.c.o.d -o CMakeFiles/testbinary.dir/__w/zephyr/zephyr/subsys/bluetooth/host/id.c.o -c /__w/zephyr/zephyr/subsys/bluetooth/host/id.c
In file included from /__w/zephyr/zephyr/include/zephyr/kernel/mm.h:17,
                 from /__w/zephyr/zephyr/include/zephyr/sys/device_mmio.h:48,
                 from /__w/zephyr/zephyr/include/zephyr/device.h:15,
                 from /__w/zephyr/zephyr/include/zephyr/drivers/bluetooth/hci_driver.h:24,
                 from /__w/zephyr/zephyr/subsys/bluetooth/host/id.c:18:
/__w/zephyr/zephyr/include/zephyr/kernel/mm/demand_paging.h:206:10: fatal error: syscalls/demand_paging.h: No such file or directory
  206 | #include <syscalls/demand_paging.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

**Environment (please complete the following information):**
 - Commit SHA: 23324fc5bae43a40024fdde05c893c75f3a73fff

--- a/arch/x86/core/userspace.c
+++ b/arch/x86/core/userspace.c
@@ -11,6 +11,10 @@
 #include <ksched.h>
 #include <x86_mmu.h>
 
+#ifdef CONFIG_DEMAND_PAGING
+#include <zephyr/kernel/mm/demand_paging.h>
+#endif
+
 #ifndef CONFIG_X86_KPTI
 /* Update the to the incoming thread's page table, and update the location of
  * the privilege elevation stack.

--- a/include/zephyr/kernel/mm.h
+++ b/include/zephyr/kernel/mm.h
@@ -14,7 +14,6 @@
 #endif
 
 #include <zephyr/kernel/internal/mm.h>
-#include <zephyr/kernel/mm/demand_paging.h>
 
 /**
  * @brief Kernel Memory Management

--- a/include/zephyr/kernel/thread.h
+++ b/include/zephyr/kernel/thread.h
@@ -8,7 +8,7 @@
 #define ZEPHYR_INCLUDE_KERNEL_THREAD_H_
 
 #ifdef CONFIG_DEMAND_PAGING_THREAD_STATS
-#include <zephyr/kernel/mm.h>
+#include <zephyr/kernel/mm/demand_paging.h>
 #endif
 
 #include <zephyr/kernel/stats.h>

--- a/include/zephyr/sys/arch_interface.h
+++ b/include/zephyr/sys/arch_interface.h
@@ -517,6 +517,8 @@ static inline unsigned int arch_num_cpus(void);
  */
 
 #ifdef CONFIG_USERSPACE
+#include <zephyr/arch/syscall.h>
+
 /**
  * Invoke a system call with 0 arguments.
  *

--- a/kernel/mmu.c
+++ b/kernel/mmu.c
@@ -20,6 +20,10 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_DECLARE(os, CONFIG_KERNEL_LOG_LEVEL);
 
+#ifdef CONFIG_DEMAND_PAGING
+#include <zephyr/kernel/mm/demand_paging.h>
+#endif
+
 /*
  * General terminology:
  * - A page frame is a page-sized physical memory region in RAM. It is a

--- a/kernel/paging/statistics.c
+++ b/kernel/paging/statistics.c
@@ -8,7 +8,7 @@
 #include <kernel_internal.h>
 #include <zephyr/internal/syscall_handler.h>
 #include <zephyr/toolchain.h>
-#include <zephyr/kernel/mm.h>
+#include <zephyr/kernel/mm/demand_paging.h>
 
 extern struct k_mem_paging_stats_t paging_stats;
 

--- a/subsys/demand_paging/backing_store/backing_store_qemu_x86_tiny.c
+++ b/subsys/demand_paging/backing_store/backing_store_qemu_x86_tiny.c
@@ -20,6 +20,7 @@
 #include <kernel_arch_interface.h>
 #include <zephyr/linker/linker-defs.h>
 #include <zephyr/sys/util.h>
+#include <zephyr/kernel/mm/demand_paging.h>
 
 void *location_to_flash(uintptr_t location)
 {

--- a/subsys/demand_paging/backing_store/ram.c
+++ b/subsys/demand_paging/backing_store/ram.c
@@ -8,6 +8,7 @@
 #include <mmu.h>
 #include <string.h>
 #include <kernel_arch_interface.h>
+#include <zephyr/kernel/mm/demand_paging.h>
 
 /*
  * TODO:

--- a/subsys/demand_paging/eviction/nru.c
+++ b/subsys/demand_paging/eviction/nru.c
@@ -10,6 +10,8 @@
 #include <kernel_arch_interface.h>
 #include <zephyr/init.h>
 
+#include <zephyr/kernel/mm/demand_paging.h>
+
 /* The accessed and dirty states of each page frame are used to create
  * a hierarchy with a numerical value. When evicting a page, try to evict
  * page with the highest value (we prefer clean, not accessed pages).

--- a/tests/kernel/fatal/exception/src/main.c
+++ b/tests/kernel/fatal/exception/src/main.c
@@ -18,6 +18,10 @@
 #include "test_syscalls.h"
 #endif
 
+#if defined(CONFIG_DEMAND_PAGING)
+#include <zephyr/kernel/mm/demand_paging.h>
+#endif
+
 #if defined(CONFIG_X86) && defined(CONFIG_X86_MMU)
 #define STACKSIZE (8192)
 #else

--- a/tests/kernel/mem_protect/demand_paging/src/main.c
+++ b/tests/kernel/mem_protect/demand_paging/src/main.c
@@ -6,6 +6,7 @@
 
 #include <zephyr/ztest.h>
 #include <zephyr/kernel/mm.h>
+#include <zephyr/kernel/mm/demand_paging.h>
 #include <zephyr/timing/timing.h>
 #include <mmu.h>
 #include <zephyr/linker/sections.h>

